Composer/packages/adaptive-form/src/components/fields/BooleanField.tsx

```diff
@@ -28,9 +28,38 @@ const BooleanField: React.FC<FieldProps> = function CheckboxWidget(props) {
     },
   ];
 
+  if (expression) {
+    options.push(
+      {
+        key: 'divider',
+        text: '-',
+        itemType: DropdownMenuItemType.Divider,
+      },
+      {
+        key: 'expression',
+        text: formatMessage('Write in expression'),
+      }
+    );
+  }
+
   const handleChange = (e, option?: IDropdownOption) => {
     if (option) {
-      const optionValue = option.key === 'none' ? undefined : option.key === 'true';
+      let optionValue: boolean | string | undefined;
```

Comment:
This looks weird, couldn't you use a union type?
what is the boolean representing?

Reply:
Those are the 3 valid types for Boolean values: true and false (boolean), expression (string) and undefined for no value.
![image](https://user-images.githubusercontent.com/66701106/96518684-41600f80-1220-11eb-8342-3c869b14382c.png)